sqlglot/dialects/doris.py

```diff
@@ -43,8 +44,89 @@ class Parser(MySQL.Parser):
             "PROPERTIES": lambda self: self._parse_wrapped_properties(),
             "UNIQUE": lambda self: self._parse_composite_key_property(exp.UniqueKeyProperty),
+            "PARTITION BY": lambda self: self._parse_partition_by_opt_range(),
         }
 
+        def _parse_partitioning_granularity_dynamic(self) -> exp.PartitionByRangePropertyDynamic:
+            self._match_text_seq("FROM")
+            start = self._parse_wrapped(self._parse_string)
+            self._match_text_seq("TO")
+            end = self._parse_wrapped(self._parse_string)
+            self._match_text_seq("INTERVAL")
+            number = self._parse_number()
+            unit = self._parse_var()
+            every = self.expression(exp.Interval, this=number, unit=unit)
+            return self.expression(
+                exp.PartitionByRangePropertyDynamic, start=start, end=end, every=every
+            )
+
+        def _parse_partition_definition(self):
+            # Handles: PARTITION `p2018` VALUES [(...), (...)), PARTITION `other` VALUES LESS THAN (MAXVALUE)
+            self._match_text_seq("PARTITION")
+            name = self._parse_id_var()
+            if self._match_text_seq("VALUES"):
+                if self._match_text_seq("LESS", "THAN"):
+                    # VALUES LESS THAN (...)
+                    value = self._parse_wrapped_csv(self._parse_expression)
+                    if isinstance(value, list) and len(value) == 1:
+                        value = value[0]
+                    return self.expression(
+                        exp.Partition,
+                        expressions=[
+                            self.expression(
+                                exp.PartitionRange,
+                                this=name,
+                                expression=value,
+                            )
+                        ],
+                    )
+                elif self._match(TokenType.L_BRACKET):
+                    # VALUES [(...), (...))
+                    values = self._parse_csv(
+                        lambda: self._parse_wrapped_csv(self._parse_expression)
+                    )
+                    self._match(TokenType.R_BRACKET)
+                    return self.expression(
+                        exp.Partition,
+                        expressions=[
+                            self.expression(
+                                exp.PartitionRange,
+                                this=name,
+                                expression=val,
+                            )
+                            for val in values
+                        ],
+                    )
+                else:
+                    self.raise_error("Unsupported VALUES syntax in PARTITION definition")
+            else:
+                self.raise_error("Expecting VALUES in PARTITION definition")
+
+        def _parse_partition_by_opt_range(
+            self,
+        ) -> exp.PartitionedByProperty | exp.PartitionByRangeProperty:
```

Comment:
Does StarRocks use the same syntax? We override this method for it as well, I wonder if we can consolidate the two.